app/controllers/account_activations_controller.rb

```diff
@@ -60,14 +60,14 @@ def resend
   private
 
   def find_user
-    @user = User.find_by!(email: params[:email], provider: @user_domain)
+    @user = User.find_by!(activation_digest: User.digest(params[:token]), provider: @user_domain)
   end
 
   def ensure_unauthenticated
     redirect_to current_user.main_room if current_user
   end
 
   def email_params
-    params.require(:email).permit(:email, :token)
+    params.require(:activation_digest).permit(:activation_digest, :token)
```

Comment:
The only param that is required is the token param. You can remove the activation_digest param